src/main/scala/ai/privado/tagger/utility/APITaggerUtility.scala

```diff
@@ -71,18 +71,25 @@ object APITaggerUtility {
           DuplicateFlowProcessor.getUniquePathsAfterDedup(flows)
       }
       apiFlows.foreach(flow => {
-        val literalCode = getLiteralCode(flow.elements.head)
-        val apiNode     = flow.elements.last
+        val sourceNode = flow.elements.head
+        val apiNode    = flow.elements.last
         // Tag API's when we find a dataflow to them
         var newRuleIdToUse = ruleInfo.id
         if (ruleInfo.id.equals(Constants.internalAPIRuleId)) addRuleTags(builder, apiNode, ruleInfo, ruleCache)
         else {
-          val domain = getDomainFromString(literalCode)
+          val domain = {
+            if (sourceNode.originalPropertyValue.nonEmpty) {
```

Comment:
? why is sourceNode.originalPropertyValue nonEmpty only for k8s domains ?

Reply:
this is to check if the domain is present inside `YAML` file or not.

Reply:
handled the domain resolution for k8s.